tap_hubspot_beta/streams.py

```diff
@@ -2532,3 +2532,27 @@ class GeolocationSummaryMonthlyStream(FormsSummaryMonthlyStream):
         th.Property("start_date", th.DateType),
         th.Property("end_date", th.DateType),
     ).to_dict() 
+
+class LeadsStream(ObjectSearchV3):
+    """Leads Stream"""
+
+    name = "leads"
+    path = "crm/v3/objects/leads/search"
+    properties_url = "crm/v3/properties/leads"
+
+    replication_key_filter = "hs_lastmodifieddate"
+
+    @cached_property
+    def schema(self):
+        properties = copy.deepcopy(self.base_properties)
+        headers = self.http_headers
+        headers.update(self.authenticator.auth_headers or {})
+        url = self.url_base + self.properties_url
+        response = self.request_decorator(self.request_schema)(url, headers=headers)
+
+        fields = response.json().get("results",[])
```

Comment:
@keyn4 if I'm not mistaken, the logic that already uses is lacking the "results" access here, which leads to failure. Can you check it please?

Reply:
I see what you mean, I do think it's cleaner if we keep the logic in one place specially due to type handling, maybe we could use v2 properties url `/properties/v2/leads/properties` or do something like `response.json().get("results",[]) if response.json.get("results") else response.json()`